File: languages/go/entity/identity/string_attribute_descriptor.go
package identity

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// StringAttributeDescriptor represents a descriptor for an attribute with a string value.
type StringAttributeDescriptor struct {
	AttributeDescriptor
}

// NewStringAttributeDescriptor creates a new StringAttributeDescriptor instance with the provided details.
func NewStringAttributeDescriptor(name string, id string, description string, required bool) StringAttributeDescriptor {
	a := AttributeDescriptor{
		DisplayName: name,
		Id:          id,
		Description: description,
		Required:    required,
	}
	return StringAttributeDescriptor{
		AttributeDescriptor: a,
	}
}

func NewStringAttributeDescriptorFromProto(s *proto.StringAttributeDefinition) StringAttributeDescriptor {
	if s == nil {
		return StringAttributeDescriptor{}
	}
	a := AttributeDescriptor{
		DisplayName: s.GetId(),
		Id:          s.Id,
		Description: s.Description,
		Required:    s.Required,
	}
	return StringAttributeDescriptor{
		AttributeDescriptor: a,
	}
}

func (s StringAttributeDescriptor) ToProto() *proto.StringAttributeDefinition {
	return &proto.StringAttributeDefinition{
		DisplayName: s.DisplayName,
		Id:          s.Id,
		Description: s.Description,
		Required:    s.Required,
	}
}

File: languages/js/src/entity/identity/string_attribute_descriptor.ts
import * as identityEntitiesProto from "../../bridge/proto/identity_entities";
import { AttributeDescriptor } from "./attribute_descriptor";

/**
 * Represents a descriptor for an attribute with a string value.
 */
export class StringAttributeDescriptor extends AttributeDescriptor {
  constructor(
    displayName: string,
    technicalName: string,
    description: string,
    required: boolean
  ) {
    super(displayName, technicalName, description, required);
  }

  public toProto(): identityEntitiesProto.StringAttributeDefinition {
    return identityEntitiesProto.StringAttributeDefinition.fromPartial({
      displayName: this.displayName,
      id: this.technicalName,
      description: this.description,
      required: this.required
    });
  }

  static fromProto(
    r: identityEntitiesProto.StringAttributeDefinition
  ): StringAttributeDescriptor {
    return new StringAttributeDescriptor(
      r.displayName,
      r.id,
      r.description,
      r.required
    );
  }
}
